paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -327,22 +321,36 @@ internal class CustomerSheetViewModel @Inject constructor(
             customerAdapter.detachPaymentMethod(
                 paymentMethodId = paymentMethod.id!!
             ).onSuccess {
-                updateViewState<CustomerSheetViewState.SelectPaymentMethod> { viewState ->
-                    viewState.copy(
-                        savedPaymentMethods = viewState.savedPaymentMethods.filter { pm ->
-                            pm.id != paymentMethod.id!!
-                        },
-                        paymentSelection = viewState.paymentSelection.takeUnless { selection ->
-                            val removedPaymentSelection = selection is PaymentSelection.Saved &&
-                                selection.paymentMethod == paymentMethod
-
-                            if (removedPaymentSelection && savedPaymentSelection == selection) {
-                                savedPaymentSelection = null
-                            }
-
-                            removedPaymentSelection
-                        } ?: savedPaymentSelection,
-                    )
+                when (val currentViewState = viewState.value) {
+                    is CustomerSheetViewState.SelectPaymentMethod -> {
+                        val savedPaymentMethods = currentViewState.savedPaymentMethods.filter { pm ->
+                            paymentMethod.id?.let { id ->
```

Comment:
This is some poor modelling from a while ago, but a payment method will have an ID. We can force this here.